### PR TITLE
[plugin.video.tv3.cat] 1.4.8

### PR DESCRIPTION
**Pull request by jqandreu**
### Description

Adapted to Kodi 19.0 and fixed errors
### Checklist:

- [x ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x ] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x ] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

**Comment by enen92** [comment]
Please have a look at travis failures, you have quite a bit of changes ahead before anyone actually review the code:
https://travis-ci.com/github/xbmc/repo-plugins/builds/218838825

File: plugin.video.tv3.cat/addon.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        <summary lang="ca">Tots els programes de TV3 a la carta.</summary>
+        <description lang="ca">Tota la programació de TV3 a la carta. [CR]Les llistes dels més vistos, destacats, programes ordenats per ordre alfabètic. [CR]Els canals en directe: TV3cat, K33, Super3, 3/24 i Esport3 </description>
+        <summary lang="es">Toda la programación de TV3 a la carta</summary>
+        <description lang="es">Toda la programación de TV3 a la carta.[CR]Las listas de los más vistos, destacados, programas ordenados de la A a la Z. [CR]Canales en directo: TV3cat, K33, Super3, 3/24, Esport3</description>
+        <language>ca</language>
+        <license>GPL-2.0</license>
+        <source>https://github.com/jqandreu/plugin.video.tv3.cat</source>
+        <website>http://www.tv3.cat</website>
+    </extension>
+    <assets>
```
Comment (enen92) [inline plugin.video.tv3.cat/addon.xml]: assets is a child of metadata not root/addon

File: plugin.video.tv3.cat/resources/language/Catalan/strings.po [hunk 2]
```diff
@@ -0,0 +1,125 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/language/Catalan/strings.po]: move to: resources/language/resource.language.ca_es/strings.po

File: plugin.video.tv3.cat/resources/language/English/strings.po [hunk 3]
```diff
@@ -0,0 +1,125 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/language/English/strings.po]: move to: resources/language/resource.language.en_gb/strings.po

File: plugin.video.tv3.cat/resources/language/German/strings.po [hunk 4]
```diff
@@ -0,0 +1,125 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/language/German/strings.po]: move to: resources/language/resource.language.de_de/strings.po

File: plugin.video.tv3.cat/resources/language/Norwegian/strings.po [hunk 5]
```diff
@@ -0,0 +1,125 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/language/Norwegian/strings.po]: move to: resources/language/resource.language.nb_no/strings.po

File: plugin.video.tv3.cat/resources/language/Spanish/strings.po [hunk 6]
```diff
@@ -0,0 +1,125 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/language/Spanish/strings.po]: move to: resources/language/resource.language.es_es/strings.po

File: plugin.video.tv3.cat/addon.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        <import addon="xbmc.python" version="3.0.0"/>
+		<import addon="script.module.beautifulsoup4" version="4.6.3"/>
+        <import addon="script.module.future" version="0.17.1"/>
+    </requires>
+    <extension point="xbmc.python.pluginsource" library="addon.py">
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <platform>all</platform>
+        <summary lang="en">Entertainment, news, documentaries, etc from catalan television www.tv3.cat</summary>
```
Comment (enen92) [inline plugin.video.tv3.cat/addon.xml]: en_GB

File: plugin.video.tv3.cat/addon.xml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        <description lang="en">Entertainment, news, documentaries, etc from catalan television www.tv3.cat. [CR]Featured shows, most viewed, A to Z programs. [CR]Live channels: TV3cat, K33, Super3, News 3/24, Esport3</description>
```
Comment (enen92) [inline plugin.video.tv3.cat/addon.xml]: en_GB

File: plugin.video.tv3.cat/addon.xml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        <summary lang="ca">Tots els programes de TV3 a la carta.</summary>
+        <description lang="ca">Tota la programació de TV3 a la carta. [CR]Les llistes dels més vistos, destacats, programes ordenats per ordre alfabètic. [CR]Els canals en directe: TV3cat, K33, Super3, 3/24 i Esport3 </description>
+        <summary lang="es">Toda la programación de TV3 a la carta</summary>
+        <description lang="es">Toda la programación de TV3 a la carta.[CR]Las listas de los más vistos, destacados, programas ordenados de la A a la Z. [CR]Canales en directo: TV3cat, K33, Super3, 3/24, Esport3</description>
```
Comment (enen92) [inline plugin.video.tv3.cat/addon.xml]: es_ES

File: plugin.video.tv3.cat/addon.xml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    <extension point="xbmc.python.pluginsource" library="addon.py">
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <platform>all</platform>
+        <summary lang="en">Entertainment, news, documentaries, etc from catalan television www.tv3.cat</summary>
+        <description lang="en">Entertainment, news, documentaries, etc from catalan television www.tv3.cat. [CR]Featured shows, most viewed, A to Z programs. [CR]Live channels: TV3cat, K33, Super3, News 3/24, Esport3</description>
+        <summary lang="ca">Tots els programes de TV3 a la carta.</summary>
+        <description lang="ca">Tota la programació de TV3 a la carta. [CR]Les llistes dels més vistos, destacats, programes ordenats per ordre alfabètic. [CR]Els canals en directe: TV3cat, K33, Super3, 3/24 i Esport3 </description>
+        <summary lang="es">Toda la programación de TV3 a la carta</summary>
```
Comment (enen92) [inline plugin.video.tv3.cat/addon.xml]: es_ES

File: plugin.video.tv3.cat/addon.xml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+        <import addon="script.module.future" version="0.17.1"/>
+    </requires>
+    <extension point="xbmc.python.pluginsource" library="addon.py">
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <platform>all</platform>
+        <summary lang="en">Entertainment, news, documentaries, etc from catalan television www.tv3.cat</summary>
+        <description lang="en">Entertainment, news, documentaries, etc from catalan television www.tv3.cat. [CR]Featured shows, most viewed, A to Z programs. [CR]Live channels: TV3cat, K33, Super3, News 3/24, Esport3</description>
+        <summary lang="ca">Tots els programes de TV3 a la carta.</summary>
```
Comment (enen92) [inline plugin.video.tv3.cat/addon.xml]: ca_ES

File: plugin.video.tv3.cat/addon.xml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+        <description lang="ca">Tota la programació de TV3 a la carta. [CR]Les llistes dels més vistos, destacats, programes ordenats per ordre alfabètic. [CR]Els canals en directe: TV3cat, K33, Super3, 3/24 i Esport3 </description>
```
Comment (enen92) [inline plugin.video.tv3.cat/addon.xml]: ca_ES

File: plugin.video.tv3.cat/addon.xml [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+        <summary lang="es">Toda la programación de TV3 a la carta</summary>
+        <description lang="es">Toda la programación de TV3 a la carta.[CR]Las listas de los más vistos, destacados, programas ordenados de la A a la Z. [CR]Canales en directo: TV3cat, K33, Super3, 3/24, Esport3</description>
+        <language>ca</language>
+        <license>GPL-2.0</license>
```
Comment (enen92) [inline plugin.video.tv3.cat/addon.xml]:
```suggestion
        <license>GPL-2.0-only</license>
```

or 

```suggestion
        <license>GPL-2.0-or-later</license>
```

File: plugin.video.tv3.cat/addon.xml (hunk 12 above)
Comment (jqandreu) [reply plugin.video.tv3.cat/addon.xml]: Fixed all errors. Thanks

**Comment by jqandreu** [comment]
Fixed all errors and sugestions. The last check also fails due to assets tag in addon.xml. I moved to extension point="xbmc.addon.metadata" Is this correct?

File: plugin.video.tv3.cat/addon.xml (hunk 1 above)
Comment (enen92) [reply plugin.video.tv3.cat/addon.xml]: You have to delete the assets that is outside of the metadata. Right now you have multiple copies

Comment (jqandreu) [reply plugin.video.tv3.cat/addon.xml]: Fixed. Thanks

File: plugin.video.tv3.cat/resources/lib/tv3cat/DirAZemisio.py [hunk 14]
```diff
@@ -0,0 +1,18 @@
+from resources.lib.video.FolderVideo import FolderVideo
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/lib/tv3cat/DirAZemisio.py:1]: dummy empty  `__init__py` files should no longer be needed with python 3

File: plugin.video.tv3.cat/resources/lib/tv3cat/TV3Strings.py [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+            'coleccions': 30023,
+            'tv3_int': 30024,
+            'canal324_int': 30025,
+            'c33super3_int': 30026,
+            'esport3_int': 30027
+        }
+
+    def get(self, string):
+        code = self.strs[string]
+        st = self.addon.getLocalizedString(code).encode("utf-8")
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/lib/tv3cat/TV3Strings.py:38]: shouldn't be needed to encode anything in python3

**Review by enen92 — APPROVED** [review]
Code looks fine, a few minors if you want to adjust but not blockers.

**Comment by jqandreu** [comment]
Ok. I will adjust in the next version. Thank you

**Pull request merged**